src/pages/gd/Swap/SwapCelo/index.tsx

```diff
@@ -1,65 +1,209 @@
-    const { connectedEnv } = useGetEnvChainId(42220)
-    const gdToken = G$ContractAddresses('GoodDollar', connectedEnv)
-    const [accountInfo, setAccountInfo] = useState<AccountInfo | undefined>(undefined)
+    const uniTheme = theme === 'dark' ? darkTheme : lightTheme
+    const { account } = useActiveWeb3React()
+    const { web3Provider } = useWeb3Context()
     const [, connect] = useConnectWallet()
+    const globalDispatch = useDispatch()
+    const { connectedEnv } = useGetEnvChainId(42220)
+    const gdTokenAddress = G$ContractAddresses('GoodDollar', connectedEnv) as string
+    const customTheme = {
+        ...uniTheme,
+        primary: '#404040',
+        fontFamily: 'Roboto',
+        accent: '#00AEFF',
+        outline: '#00AFFF',
+        active: '#00AFFF',
+        // onAccent: '#00AEFF',
+        accentSoft: '#00AEFF',
+        networkDefaultShadow: 'hsl(199deg 100% 50% / 20%)',
+    }
+
+    const gdToken = {
+        chainId: 42220,
+        address: gdTokenAddress,
+        name: connectedEnv.includes('production') ? 'GoodDollar' : 'GoodDollar Dev',
+        symbol: 'G$',
+        decimals: 18,
+        logoURI:
+            'https://raw.githubusercontent.com/GoodDollar/GoodProtocolUI/master/src/assets/images/tokens/gd-logo.png',
+    }
+
+    celoTokenList.push(gdToken)
+
+    const connectOnboard = useCallback(async () => {
+        if (!account) {
+            // todo: make connect onboard a generic function/merge with: useOnboardConnect
+            const osName = getDevice().os.name
+            // temp solution for where it tries and open a deeplink for desktop app
+            if (['Linux', 'Windows', 'macOS'].includes(osName)) {
+                AsyncStorage.safeRemove('WALLETCONNECT_DEEPLINK_CHOICE')
+            }
 
-    useEffect(() => {
-        if (account && library) {
-            setAccountInfo({
-                account: account,
-                explorerUrl: process.env.REACT_APP_CELO_RPC ?? 'https://forno.celo.org',
-                chainId: 42220,
-                provider: library,
-            })
+            const connected = await connect()
+            if (!connected) {
+                return false
+            }
         }
-    }, [account, library])
+        return true
+    }, [connect])
+
+    const handleError = useCallback(async (e) => {
+        console.log('handleError -->', { e })
+    }, [])
+
+    const handleTxFailed: OnTxFail = useCallback(async (error: string, data: any) => {
+        console.log('handleTxFailed -->', { error, data })
+    }, [])
+
+    const handleTxSubmit: OnTxSubmit = useCallback(
+        async (txHash: string, data: any) => {
+            console.log('handleTxSubmit -->', { txHash, data })
+            // todo: get info from receipt data, below is just placeholder
+            const { info } = data
+            switch (info.type) {
+                //approve
+                case 0: {
+                    // do we want to add this to transaction list as well?
```

Comment:
yes why not?

Reply:
ok done
just not sure where to get symbol X from @sirpy 
for now just using Approved spending